diff -uNr mc-4.8.32/lib/unixcompat.h mc-4.8.32.mod/lib/unixcompat.h
--- mc-4.8.32/lib/unixcompat.h	2024-08-08 15:49:18.000000000 +0800
+++ mc-4.8.32.mod/lib/unixcompat.h	2024-08-26 11:08:36.408859679 +0800
@@ -108,7 +108,7 @@
 #define PATH_SEP_STR "/"
 #define IS_PATH_SEP(c) ((c) == PATH_SEP)
 #define PATH_ENV_SEP ':'
-#define TMPDIR_DEFAULT "/tmp"
+#define TMPDIR_DEFAULT "@TERMUX_PREFIX@/tmp"
 #define SCRIPT_SUFFIX ""
 #define get_default_editor() "vi"
 #define OS_SORT_CASE_SENSITIVE_DEFAULT TRUE
