diff -uNr dar-2.7.14/Makefile.in dar-2.7.14.mod/Makefile.in
--- dar-2.7.14/Makefile.in      2024-03-23 19:43:35.000000000 +0800
+++ dar-2.7.14.mod/Makefile.in  2024-04-04 14:16:46.319993011 +0800
@@ -144,12 +144,12 @@
 am__v_at_1 = 
 SOURCES =
 DIST_SOURCES =
-RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
+RECURSIVE_TARGETS = all-recursive cscopelist-recursive \
        ctags-recursive dvi-recursive html-recursive info-recursive \
        install-data-recursive install-dvi-recursive \
        install-exec-recursive install-html-recursive \
        install-info-recursive install-pdf-recursive \
-       install-ps-recursive install-recursive installcheck-recursive \
+       install-ps-recursive install-recursive \
        installdirs-recursive pdf-recursive ps-recursive \
        tags-recursive uninstall-recursive
 am__can_run_installinfo = \
@@ -715,9 +715,7 @@
            --srcdir=../.. --prefix="$$dc_install_base" \
          && $(MAKE) $(AM_MAKEFLAGS) \
          && $(MAKE) $(AM_MAKEFLAGS) $(AM_DISTCHECK_DVI_TARGET) \
-         && $(MAKE) $(AM_MAKEFLAGS) check \
          && $(MAKE) $(AM_MAKEFLAGS) install \
-         && $(MAKE) $(AM_MAKEFLAGS) installcheck \
          && $(MAKE) $(AM_MAKEFLAGS) uninstall \
          && $(MAKE) $(AM_MAKEFLAGS) distuninstallcheck_dir="$$dc_install_base" \
                distuninstallcheck \
@@ -765,8 +763,6 @@
          || { echo "ERROR: files left in build directory after distclean:" ; \
               $(distcleancheck_listfiles) ; \
               exit 1; } >&2
-check-am: all-am
-check: check-recursive
 all-am: Makefile $(DATA) config.h
 installdirs: installdirs-recursive
 installdirs-am:
@@ -778,7 +774,6 @@
 install-am: all-am
        @$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
 
-installcheck: installcheck-recursive
 install-strip:
        if test -z '$(STRIP)'; then \
          $(MAKE) $(AM_MAKEFLAGS) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
@@ -848,8 +843,6 @@
 
 install-ps-am:
 
-installcheck-am:
 
 maintainer-clean: maintainer-clean-recursive
        -rm -f $(am__CONFIG_DISTCLEAN_FILES)
@@ -873,7 +866,7 @@
 .MAKE: $(am__recursive_targets) all install-am install-strip
 
 .PHONY: $(am__recursive_targets) CTAGS GTAGS TAGS all all-am \
-       am--refresh check check-am clean clean-cscope clean-generic \
+       am--refresh clean clean-cscope clean-generic \
        clean-libtool clean-local cscope cscopelist-am ctags ctags-am \
        dist dist-all dist-bzip2 dist-gzip dist-lzip dist-shar \
        dist-tarZ dist-xz dist-zip dist-zstd distcheck distclean \
@@ -884,7 +877,7 @@
        install-exec install-exec-am install-html install-html-am \
        install-info install-info-am install-man install-pdf \
        install-pdf-am install-ps install-ps-am install-strip \
-       installcheck installcheck-am installdirs installdirs-am \
+       installdirs installdirs-am \
        maintainer-clean maintainer-clean-generic mostlyclean \
        mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
        tags tags-am uninstall uninstall-am
