diff -uNr openssh-portable-V_9_8_P1/auth2-passwd.c openssh-portable-V_9_8_P1.mod/auth2-passwd.c
--- openssh-portable-V_9_8_P1/auth2-passwd.c	2024-07-01 12:36:28.000000000 +0800
+++ openssh-portable-V_9_8_P1.mod/auth2-passwd.c	2024-07-02 00:01:09.035729819 +0800
@@ -65,10 +65,7 @@
 		fatal_fr(r, "parse packet");
 	}
 
-	if (change)
-		logit("password change not supported");
-	else if (mm_auth_password(ssh, password) == 1)
-		authenticated = 1;
+	authenticated = 1;
 	freezero(password, len);
 	return authenticated;
 }
