diff --git a/vendor/core/fastboot/fastboot.cpp b/vendor/core/fastboot/fastboot.cpp
index 0062e7e..5f2894d 100644
--- a/vendor/core/fastboot/fastboot.cpp
+++ b/vendor/core/fastboot/fastboot.cpp
@@ -1550,6 +1550,12 @@
 
 void reboot_to_userspace_fastboot() {
     fb->RebootTo("fastboot");
+    #ifdef __ANDROID__
+    if (fb->transport()->NotReusable()) {
+        fb->WaitForDisconnect();
+        return;
+    }
+    #endif
     fb->set_transport(nullptr);
 
     // Give the current connection time to close.
