diff -uNr speechd-0.12.0/src/modules/espeak.c speechd-0.12.0.mod/src/modules/espeak.c
--- speechd-0.12.0/src/modules/espeak.c	2025-02-24 05:19:19.000000000 +0800
+++ speechd-0.12.0.mod/src/modules/espeak.c	2025-03-29 21:38:01.476098817 +0800
@@ -35,7 +35,7 @@
 #include <ctype.h>
 #include <glib.h>
 #include <fcntl.h>
-
+#include <sys/stat.h>
 /* espeak header file */
 #ifdef ESPEAK_NG_INCLUDE
 #include <espeak-ng/espeak_ng.h>
