diff -uNr swtpm-0.8.2/configure.ac swtpm-0.8.2.mod/configure.ac
--- swtpm-0.8.2/configure.ac    2024-03-23 04:52:59.000000000 +0800
+++ swtpm-0.8.2.mod/configure.ac        2024-04-04 14:25:35.683992809 +0800
@@ -381,15 +381,6 @@
 
 have_tcsd=no
 AS_IF([test "$enable_tests" = "yes"],[
-       dnl We either need netstat (more common across systems) or 'ss' for test cases
-       AC_PATH_PROG([NETSTAT], [netstat])
-       if test "x$NETSTAT" = "x"; then
-               AC_PATH_PROG([SS], [ss])
-               if test "x$SS" = "x"; then
-                       AC_MSG_ERROR(['netstat' and 'ss' tools are missing for tests: net-tools OR iproute/iproute2 package])
-               fi
-       fi
-
        AC_PATH_PROG([TCSD], tcsd)
        if test "x$TCSD" = "x"; then
                have_tcsd=no
@@ -398,21 +389,11 @@
                have_tcsd=yes
        fi
 
-       AC_PATH_PROG([EXPECT], expect)
-       if test "x$EXPECT" = "x"; then
-               AC_MSG_ERROR([expect is required: expect package])
-       fi
-
        AC_PATH_PROG([GAWK], gawk)
        if test "x$GAWK" = "x"; then
                AC_MSG_ERROR([gawk is required: gawk package])
        fi
 
-       AC_PATH_PROG([SOCAT], socat)
-       if test "x$SOCAT" = "x"; then
-               AC_MSG_ERROR([socat is required: socat package])
-       fi
-
        AC_PATH_PROG([BASE64], base64)
        if test "x$BASE64" = "x"; then
                AC_MSG_ERROR([base64 is required: base64 package])
