diff -uNr uwsgi-2.0.25/core/uwsgi.c uwsgi-2.0.25.mod/core/uwsgi.c
--- uwsgi-2.0.25/core/uwsgi.c   2024-04-14 03:02:37.000000000 +0800
+++ uwsgi-2.0.25.mod/core/uwsgi.c       2024-04-14 14:06:43.729158656 +0800
@@ -2690,7 +2690,7 @@
 #ifdef __APPLE__
                        uwsgi_string_new_list(&uwsgi.mime_file, "/etc/apache2/mime.types");
 #else
-                       uwsgi_string_new_list(&uwsgi.mime_file, "/etc/mime.types");
+                       uwsgi_string_new_list(&uwsgi.mime_file, "@TERMUX_PREFIX@/etc/mime.types");
 #endif
                struct uwsgi_string_list *umd = uwsgi.mime_file;
                while (umd) {
