diff -uNr wezterm-20240203-110809-5046fc22/wezterm-gui/build.rs wezterm-20240203-110809-5046fc22.mod/wezterm-gui/build.rs
--- wezterm-20240203-110809-5046fc22/wezterm-gui/build.rs	2024-02-04 02:08:09.000000000 +0800
+++ wezterm-20240203-110809-5046fc22.mod/wezterm-gui/build.rs	2025-01-19 22:50:00.534245887 +0800
@@ -1,4 +1,8 @@
 fn main() {
+    println!("cargo:rustc-link-search=native=$PREFIX/lib");
+    println!("cargo:rustc-link-lib=dylib=EGL");
+    println!("cargo:rustc-link-lib=dylib=nativewindow");
+    println!("cargo:rustc-link-lib=dylib=GLESv2");
     println!("cargo:rerun-if-changed=build.rs");
 
     #[cfg(windows)]
