diff -uNr runc-1.2.0/libcontainer/dmz/cloned_binary_linux.go runc-1.2.0.mod/libcontainer/dmz/cloned_binary_linux.go
--- runc-1.2.0/libcontainer/dmz/cloned_binary_linux.go	2024-10-22 06:21:47.000000000 +0800
+++ runc-1.2.0.mod/libcontainer/dmz/cloned_binary_linux.go	2024-10-26 20:05:32.707522936 +0800
@@ -133,10 +133,8 @@
 	tmpDirs := []string{
 		tmpDir,
 		os.TempDir(),
-		"/tmp",
+		"@TERMUX_PREFIX@/tmp",
 		".",
-		"/bin",
-		"/",
 	}
 
 	// Try to fallback to O_TMPFILE (supported since Linux 3.11).
